app/views/LicensorViewComponent.Vue

```diff
@@ -0,0 +1,392 @@
+        .form-group
+            label.small
+            | Token Url
+            =" "
+            input(type="text" name="tokenUrl")
+        .form-group
+            label.small
+            | Token Method
+            =" "
+            select(name="tokenMethod")
+                option(value='') None
+                option(value="get") GET
+                option(value="post") POST
+        .form-group
+            label.small
+            | Token Auth User (Our client ID to access token url)
+            =" "
+            input(type="text" name="tokenAuthUser")
+        .form-group
+            label.small
+            | Token Auth Password (Our client password to access token url)
+            =" "
+            input(type="text" name="tokenAuthPass")
+        .form-group
+            label.small
+            | Strict SSL
+            =" "
+            select(name="strictSSL")
+                option(value='') None
+                option(value="true") True
+                option(value="false") False
+        .form-group
+            label.small
+            | Redirect Url After Login
+            =" "
+            input(type="text" name="redirectAfterLogin")
+        .form-group
+            button.btn.btn-primary(v-on:click.prevent="onCreateOauth", name="createProvider") Create Provider
+            button.btn.btn-primary(v-on:click.prevent="onEditOauth", name="editProvider") Edit Provider
+            h4.small *Edit will not remove any values that you leave blank here, it will only update if you populate any field
+
+    h3 Show OAuthProvider
+    form#oauth-show-form
+        .form-group
+            label.small
+            | Provider name
+            =" "
+            input(type="text", name="oauthNameShow")
+        .form-group
+            button.btn.btn-primary(v-on:click.prevent="onShowOauth", name="showProvider") Show Provider
+    table(v-for = "oauth in oauthProvider")
+      .table.table-condensed#o-auth-table
+        tr
+            th.border(v-for="(value, key) in oauth") {{key}}
+        tr
+            td.border(v-for="(value, key) in oauth") {{value}}
+  
+</template>
+
+<script lang="coffee">
+co = require('co')
+api = require 'core/api'
+moment = require('moment')
+Prepaids = require 'collections/Prepaids'
+forms = require 'core/forms'
+
+module.exports = Vue.extend({
+  data: ->
+    prepaids: []
+    clients: []
+    oauthProvider: []
+
+  methods:
+    runValidation: (element, requiredProps) ->
+        forms.clearFormAlerts(element)
+        data = forms.formToObject(element[0])
+        result = tv4.validateMultiple(data, {required: requiredProps})
+        unless result.valid
+            forms.applyErrorsToForm(element, result.errors)
+            return
+        return data
+
+    onCreateLicense: co.wrap ->
+        el = $('#prepaid-form')
+        requiredProps = ['email', 'maxRedeemers','startDate', 'endDate']
+        data = @runValidation(el, requiredProps)
+        unless data
+            return
```

Comment:
Assumes validation errors, which updates the UI to indicate error internally to `runValidation`?

Reply:
@differentmatt I did not understand this question, can you explain?

Reply:
The following checks use `forms.setErrorToProperty`, but this one doesn't.  I was wondering why.

Reply:
Inside `runValidation`, it uses `forms.applyErrorstoForm` to apply multiple errors at once, which internally calls `setErrorToProperty`
The following checks just apply a single error, thats why used `forms.setErrorToProperty` directly